### ISSUE TITLE
Support custom workload identity pools

### ISSUE THREAD

**Issue by AndrewBurian** [GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp#206]
### TL;DR
The plugin makes the assumption that you are using the default GKE workload identity pool, which may not always be the case.

Fetch the pool from the cluster API instead.

### Design

**Proposal**
Right now, the workload ID pool used by the gcp provider is hard coded to always be [`"%s.svc.id.goog"` ](https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/blob/v1.1.0/auth/auth.go#L102), which was a safe optimization when that was originally written. 

Now GKE clusters support [alternate workload pool configurations](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#workloadidentityconfig). A cluster using an alternate pool will cause this provider to fail when the ID token requested from the kube-api has an mismatched audience to the pool.

The "simple" fix is to use the other metadata collected in the token step to fetch the cluster details from the google APIs, which will include the pool id used in the cluster. I suspect this is not actually simple since it means adding an authenticated call to the googleapis coming directly from the plugin, which means it itself needs to be able to make use of a workload identity, and would be a breaking change for all existing deployments.

**Alternatives considered**
I couldn't find any details on what exactly the GKE metadata server has set as values, but if there's another way to source this information about the cluster from it, that would be better.

**Resources**
- https://cloud.google.com/iam/docs/reference/sts/rest/v1/TopLevel/token
- https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#workloadidentityconfig
- https://cloud.google.com/iam/docs/reference/rpc/google.iam.v2#google.iam.v2.DenyRule
- https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/blob/v1.1.0/auth/auth.go#L102

**Additional information**
This is currently out of the ordinary for a GKE cluster to use a custom defined workload federation pool, but I strongly suspect it will become a rising use case.

The GCP ecosystem is gradually adopting [iam.v2 principal identifiers](https://cloud.google.com/iam/docs/reference/rpc/google.iam.v2#google.iam.v2.DenyRule), which when implemented will allow workload identities to directly be granted permissions in GCP without the service account impersonation step. As that develops, more control over the workload federation pools will be a common request and more clusters will use customer-defined pools.
